### ISSUE TITLE
libGeomPainter is not loaded before libROBAST

### ISSUE THREAD

**Issue by zorba-the-geek** [ROBAST/ROBAST#16]
Can't find info on this issue:

> root [0]  const double cm = AOpticsManager::cm()
> cling::DynamicLibraryManager::loadLibrary(): /home/user/Desktop/lunar_telescopes/software/ROBAST/libROBAST.so: undefined symbol: _ZTI8TGeoBBox
> Error in TInterpreter::TCling::AutoLoad: failure loading library libROBAST.so for AOpticsManager
> cling::DynamicLibraryManager::loadLibrary(): /home/user/Desktop/lunar_telescopes/software/ROBAST/libROBAST.so: undefined symbol: _ZTI8TGeoBBox
> Error in TInterpreter::TCling::AutoLoad: failure loading library libROBAST.so for AOpticsManager
> ROOT_prompt_0:1:19: error: 'AOpticsManager' is not a class, namespace, or enumeration
> const double cm = AOpticsManager::cm()
>                   ^
> ROOT_prompt_0:1:19: note: 'AOpticsManager' declared here

root tutorials, like the ALICE one mentioned in the guide, work just fine.

Should I compile root from source with specific flags? I had installed root binary for ubuntu 14.04 from cern website.

**Comment by akira-okumura** [comment]
Thank you for the report.

It looks that libROBAST cannot find TGeoBBox from any other dependent libraries, which should exist in libGeom.so.

Please give me some information below.
1. ROOT version.
2. The standard output of `ls $ROOTSYS/lib/libGeom*`
3. Try `root [0] gSystem->Load("libGeom")`

**Comment by zorba-the-geek** [comment]
1. root version 6.06/06  (i don't mind getting a different version)
2. 

> /home/user/Desktop/lunar_telescopes/software/root/lib/libGeomBuilder_rdict.pcm
/home/user/Desktop/lunar_telescopes/software/root/lib/libGeomBuilder.rootmap
/home/user/Desktop/lunar_telescopes/software/root/lib/libGeomBuilder.so
/home/user/Desktop/lunar_telescopes/software/root/lib/libGeomPainter_rdict.pcm
/home/user/Desktop/lunar_telescopes/software/root/lib/libGeomPainter.rootmap
/home/user/Desktop/lunar_telescopes/software/root/lib/libGeomPainter.so
/home/user/Desktop/lunar_telescopes/software/root/lib/libGeom_rdict.pcm
/home/user/Desktop/lunar_telescopes/software/root/lib/libGeom.rootmap
/home/user/Desktop/lunar_telescopes/software/root/lib/libGeom.so

3. Same error:

root [0] gSystem->Load("libGeom");
root [1]  const double cm = AOpticsManager::cm()
cling::DynamicLibraryManager::loadLibrary(): /home/zorba/Desktop/lunar_telescopes/software/ROBAST/libROBAST.so: undefined symbol: _ZTI9TGeoTrack
Error in <TInterpreter::TCling::AutoLoad>: failure loading library libROBAST.so for AOpticsManager
cling::DynamicLibraryManager::loadLibrary(): /home/zorba/Desktop/lunar_telescopes/software/ROBAST/libROBAST.so: undefined symbol: _ZTI9TGeoTrack
Error in <TInterpreter::TCling::AutoLoad>: failure loading library libROBAST.so for AOpticsManager
ROOT_prompt_1:1:19: error: 'AOpticsManager' is not a class, namespace, or enumeration
const double cm = AOpticsManager::cm()
                  ^
ROOT_prompt_1:1:19: note: 'AOpticsManager' declared here

**Comment by akira-okumura** [comment]
Will you tell me the link to the binary you downloaded? I usually build ROOT from source and have not fully checked if ROBAST works OK with a binary distribution.

**Comment by zorba-the-geek** [comment]
Thanks for the help, by the way.

Here is ubuntu 14.04 64bit download:
https://root.cern.ch/download/root_v6.06.06.Linux-ubuntu14-x86_64-gcc4.8.tar.gz

root and all tutorials work, robast not so much.

I'll try compiling root now, just got down getting dependencies.

**Comment by zorba-the-geek** [comment]
I installed that, and added the source flie to patch through environments:

.bashrc:
```

. /home/user/Desktop/lunar_telescopes/software/root/bin/thisroot.sh
export LD_LIBRARY_PATH=/home/user/Desktop/lunar_telescopes/software/ROBAST:$LD_LIBRARY_PATH
```

**Comment by akira-okumura** [comment]
OK. I will first test with a Scientific Linux 7 virtual machine. If it works, I will have to make an Ubuntu VM. The next reply will take a few hours at least, so please take a rest if your timezone is midnight or later :-)

**Comment by zorba-the-geek** [comment]
No problem, of course. I'll report back when compile is done, and if that works. I had installed all optional libraries before compiling.

**Comment by akira-okumura** [comment]
A quick workaround.
```
root [0] gSystem->Load("libGeomPainter")
root [1] root [1] double cm = AOpticsManager::mm()
```

I think something was changed in recent ROOT and library dependency check in ROBAST fails. I will update `Makefile`, but please use the above workaround for the moment.

**Comment by zorba-the-geek** [comment]
Awesome, thanks!

![out](https://cloud.githubusercontent.com/assets/6095735/18374815/a430bd40-7615-11e6-9532-9d09b831b0ab.png)

**Comment by akira-okumura** [comment]
Can you please check v2.4.3? I have updated Makefile.

**Comment by zorba-the-geek** [comment]
Sorry, I don't often use this account, I need to update my email on it.

I downloaded https://root.cern.ch/download/root_v6.09.02.Linux-ubuntu16-x86_64-gcc5.4.tar.gz

Extracted and tutorial of geomAlice.C worked just fine.

Deleted old ROBAST, git clone ROBAST, then did make, all successful.

Then go to tutorials folder run root. `.x HESS1.C+`

Output is: 

>Info in <TUnixSystem::ACLiC>: creating shared library /home/user/git/ROBAST/tutorials/./HESS1_C.so
> In file included from input_line_11:9:
> ././HESS1.C:14:10: fatal error: 'AFocalSurface.h' file not found
> #include "AFocalSurface.h"
>          ^
> Error in <ACLiC>: Dictionary generation failed!

I'll try working through it.

**Comment by zorba-the-geek** [comment]
Ok, I commented out lines 57 & 58 in HESS1.C commit c1fdd26 and that worked
` 57 //  manager->SetMultiThread(kTRUE); // enable multi threading
 58 //  manager->SetMaxThreads(8); // 8 threads
`
So yea, it does work without the workaround!

**Comment by akira-okumura** [comment]
Thank you. It looks that `TGeoManager::SetMultiThread` has been deleted in ROOT 6.09/02. That is why you had to commented out line 57.

**Comment by akira-okumura** [comment]
I will fix it in issue #22

**Comment by akira-okumura** [comment]
Fixed in a8118ed5f8f2ffb0045be3a3c6805fa1ce4daa8b
